types/node/util.promisify.d.ts

```diff
@@ -0,0 +1,48 @@
+declare module 'util' {
+    // Custom promisify must exist or promisifying functions with overloads will break
+    function promisify<TCustom extends Function>(fn: CustomPromisify<TCustom>): TCustom;
+    function promisify<T extends (...args: any[]) => any>(fn: T): Promisify<T>;
+}
+
+/**
+ * Returns a promisified function signature for the given callback-style function.
+ */
+type Promisify<
```

Comment:
`Promisify`, `Last`, `LastArgument`, `CallbackAPIReturnType`, and `Lead` probably shouldn't be global types. I would recommend placing them inside the `util` module above, instead.

Reply:
Can do that. Is there a way to make it so they cannot be imported from the `util` module?

Reply:
You should be able to do:
```ts
export {}
type Last = ...;
type LastArgument = ...;
type Lead = ...;
type Promisify = ...;

declare module 'util' {
	function promisify(...): ...;
}
```